src/App.tsx

```diff
@@ -9,22 +9,46 @@ import { Page } from './pages.ts';
 
 import './App.css';
 
-const pageComponents = {
-    [Page.ABOUT]: <About />,
-    [Page.PRODUCTS]: <Products />,
-} as const;
-
 function App() {
     const [currentPage, setCurrentPage] = useState<Page>(Page.PRODUCTS);
+    const [isDarkTheme, setIsDarkTheme] = useState(() => {
+        const savedTheme = localStorage.getItem('theme');
+        return savedTheme ? JSON.parse(savedTheme) : window.matchMedia('(prefers-color-scheme: dark)').matches;
+    });
+
+    useEffect(() => {
+        const mediaQuery = window.matchMedia('(prefers-color-scheme: dark)');
+        const handleChange = () => {
+            setIsDarkTheme(mediaQuery.matches);
+        };
+        mediaQuery.addEventListener('change', handleChange);
+        return () => {
+            mediaQuery.removeEventListener('change', handleChange);
+        };
+    }, []);
+
+    useEffect(() => {
+        document.body.className = isDarkTheme ? 'dark-theme' : 'light-theme';
```

Comment:
use constant or enum for 'dark-theme' : 'light-theme', because u are use it in several place of your app